### PR TITLE
[Fast-reboot]: Refactor fast-reboot script. Store the dumps in /host

### PR DESCRIPTION
**Pull request by pavel-shirshov**
**- What I did**
1. Refactor fast-reboot script
2. Store the dumps in /host directory to better work with NOS-SONiC updates

**- How I did it**
Modify fast-reboot and fast-reboot.py to better serve our needs

**- How to verify it**
1. Run mkdir your_dir; sudo /usr/bin/fast-reboot-dump.py your_dir to find the dumps in your_dir
2. Run sudo fast-reboot to find the fast-reboot working

**- Previous command output (if the output of a command-line utility has changed)**

**- New command output (if the output of a command-line utility has changed)**

-->

### CODE REVIEW

**Comment by lguohan** [comment]
is this also for sonic to sonic upgrade?

**Comment by pavel-shirshov** [comment]
Yes

File: scripts/fast-reboot-dump.py [hunk 1]
```diff
@@ -244,18 +245,17 @@ def generate_default_route_entries(filename):
...
+        json.dump(default_routes_output, fp, indent=2, separators=(',', ': '))
 
 
 def main():
-    all_available_macs, map_mac_ip_per_vlan = generate_fdb_entries('/tmp/fdb.json')
-    arp_entries = generate_arp_entries('/tmp/arp.json', all_available_macs)
-    generate_default_route_entries('/tmp/default_routes.json')
+    root_dir = '/tmp'
+    if len(sys.argv) > 1 and os.path.isdir(sys.argv[1]):
+        root_dir = sys.argv[1]
```
Comment (lguohan) [inline scripts/fast-reboot-dump.py]: it is better to use argparser to add option, easy to use, give you options, and helps.

https://docs.python.org/2/howto/argparse.html

Comment (pavel-shirshov) [reply scripts/fast-reboot-dump.py]: I agree

**Pull request merged**